subgraphs/euler-finance/src/mappings/handlers.ts

```diff
@@ -1,170 +1,316 @@
+  protocol.save();
+
+  market.totalDepositBalanceUSD = newTotalDepositBalanceUSD;
+  market.totalBorrowBalanceUSD = newTotalBorrowBalanceUSD;
+  market.totalValueLockedUSD = market.totalDepositBalanceUSD;
+
+  market.inputTokenBalance = totalDepositBalance;
+  market.outputTokenSupply = totalBalances;
+  market._totalBorrowBalance = totalBorrowBalance;
+
+  const dTokenAddress = Address.fromString(market._dToken!);
+  const dToken = ERC20.bind(dTokenAddress);
+  // these should always equal
+  // totalBorrows == dTokenTotalSupply
+  if (totalBorrows.gt(BIGINT_ZERO)) {
+    const dTokenTotalSupply = dToken.totalSupply();
+    if (dTokenTotalSupply.gt(BIGINT_ZERO)) {
+      market._dTokenExchangeRate = bigIntToBDUseDecimals(totalBorrowBalance, token.decimals).div(
+        bigIntToBDUseDecimals(dTokenTotalSupply, DEFAULT_DECIMALS),
+      );
+    }
+  }
+  market.save();
+
+  if (interestRate.notEqual(assetStatus.interestRate)) {
+    // update interest rates if `interestRate` or `reserveFee` changed
+    updateInterestRates(market, interestRate, assetStatus.reserveFee, totalBorrows, totalBalances, event);
+  }
+  updateRevenue(reserveBalance, totalBalances, totalBorrows, protocol, market, assetStatus, event);
+  snapshotMarket(event.block, marketId, BIGDECIMAL_ZERO, null);
+
+  //for verification only
+  const eTokenAddress = Address.fromString(market.outputToken!);
+  const eToken = ERC20.bind(eTokenAddress);
+  const eTokenTotalSupply = eToken.totalSupply();
+  if (totalBalances.notEqual(eTokenTotalSupply)) {
+    log.warning("[logAssetStatus]market={}/{},totalBalances={},eTokenTotalSupply={}", [
+      market.name!,
+      market.id,
+      totalBalances.toString(),
+      eTokenTotalSupply.toString(),
+    ]);
+  }
+
+  // update prices, tvl, totalDepositBalanceUSD, totalBorrowBalanceUSD every 100 blocks (~20 min)
+  if (event.block.number.ge(protocol._lastUpdateBlockNumber!.plus(BIGINT_HUNDRED))) {
+    const eulerContract = Euler.bind(Address.fromString(EULER_ADDRESS));
+    const execProxyAddress = eulerContract.moduleIdToProxy(MODULEID__EXEC);
+    let totalDepositBalanceUSD = BIGDECIMAL_ZERO;
+    let totalBorrowBalanceUSD = BIGDECIMAL_ZERO;
+    for (let i = 0; i < protocol._marketIDs!.length; i++) {
+      const mrktID = protocol._marketIDs![i];
+      const mrkt = getOrCreateMarket(mrktID);
+      const tkn = getOrCreateToken(Address.fromString(mrkt.inputToken));
+      const currPriceUSD = updatePrices(execProxyAddress, mrkt, event);
+      const underlyingPriceUSD = currPriceUSD ? currPriceUSD : mrkt.inputTokenPriceUSD;
+      // mark-to-market
+      mrkt.totalDepositBalanceUSD = bigIntToBDUseDecimals(mrkt.inputTokenBalance, tkn.decimals).times(
+        underlyingPriceUSD,
+      );
+      mrkt.totalBorrowBalanceUSD = bigIntToBDUseDecimals(mrkt._totalBorrowBalance!, tkn.decimals).times(
+        underlyingPriceUSD,
+      );
+      mrkt.totalValueLockedUSD = mrkt.totalDepositBalanceUSD;
+      mrkt.save();
+
+      log.debug(
```

Comment:
again, I am guessing this was for dev? Do you think it should be included in the master version?

Reply:
My preference is to keep them - users can filter what log level to show. Having them will help debug issues in the future.